### PR TITLE
Add `soak bootstrap` and `soak install` commands

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Creating a soak test AWS account, creating the cluster and bootstrapping all of the services into that cluster required an [extensive process](https://github.com/aws-controllers-k8s/test-infra/tree/main/soak). This process was well defined, but wordy and time consuming. 

This pull request introduces two new commands, `ackdev soak bootstrap` and `ackdev soak install`, which bootstraps an AWS account (and creates an EKS cluster) and installs the soak testing manifests and charts onto the cluster. This will greatly reduce the number of steps required by service teams to onboard their soak test cluster. 

The [soak test installations instructions](https://github.com/aws-controllers-k8s/test-infra/tree/main/soak) will be updated to use this new functionality. The service teams will still need to build and upload the soak test Docker image into the ECR public repository manually, but most other steps will be covered by these commands.

Example of `ackdev soak bootstrap`:
```
$ ackdev soak bootstrap --service s3
Bootstrapping ECR Public Repo ... 👍 (public.ecr.aws/<redacted>/ack-s3-soak)
Bootstrapping EKS Cluster (this may take >30 minutes) ... 👍
```

Example of `ackdev soak install`:
```
$ ackdev soak install --service s3 --image-tag v0.0.1
Adding Prometheus Helm chart repository ... 👍
Installing ACK s3 controller Helm chart ... 👍
Installing Prometheus Helm chart ... 👍
Applying ACK Grafana dashboard ... 👍
Installing soak test runner Helm chart ... 👍
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/ackdev/cmd/soak_bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	fmt.Printf("Bootstrapping ECR Public Repo ... ")
+
+	repoUri, err := soak.EnsureECRRepository(optSoakService)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("👍 (%s)\n", repoUri)
```
Comment (jaypipes) [inline cmd/ackdev/cmd/soak_bootstrap.go:68]: Here and throughout, can we please not use emojis and instead just use the word "ok"?

Yes, I know...

bitmoji

File: pkg/soak/ecr.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/aws-controllers-k8s/dev-tools/pkg/util"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecrpublic"
+)
+
+var (
+	ecrPublicRegion    = "us-east-1"
```
Comment (jaypipes) [inline pkg/soak/ecr.go:13]: yes... "global".

File: pkg/util/aws.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package util
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func NewSession() *session.Session {
+	return NewSessionWithRegion("us-west-2")
```
Comment (jaypipes) [inline pkg/util/aws.go:9]: I suppose it's fine to hard-code soak tests into us-west-2, but might be more future-proof to make this a variable. Could add in a future PR, though, of course...

File: cmd/ackdev/cmd/soak_install.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+			},
+		},
+		"aws": map[string]interface{}{
+			"region": defaultControllerRegion,
+		},
+		"serviceAccount": map[string]interface{}{
+			"create": false,
+			"name":   soak.GetDefaultServiceAccountName(service),
+		},
+	}
```
Comment (jaypipes) [inline cmd/ackdev/cmd/soak_install.go:149]: this and the below for the prometheus chart have a good chance of becoming out of sync with changes made to the test-infra/soak/ values.yaml files.

**Review by jaypipes — COMMENTED** [review]
there's a lot of (good) stuff to unpack here, @RedbackThomson. My primary concern is that because this is in the dev-tools `ackdev` CLI tool instead of in test-infra's `soak/` directory, there's a high likelihood that changes to test-infra/soak's values.yaml files or kustomizations are going to not get picked up here and cause things to get out of sync. I worry that creating a Go CLI command to automate a bunch of Bash commands will be hard to maintain in the long-run.

File: cmd/ackdev/cmd/soak_bootstrap.go (hunk 1 above)
Comment (RedbackThomson) [reply cmd/ackdev/cmd/soak_bootstrap.go:68]: I knew you'd call me out on this before I even posted the pull request. I kind of did it just to see your response. 😆

Comment (jaypipes) [reply cmd/ackdev/cmd/soak_bootstrap.go:68]: LOL :)

**Comment by RedbackThomson** [comment]
Closing in favour of https://github.com/aws-controllers-k8s/test-infra/pull/182

**Pull request closed without merging**